### PR TITLE
Feature tour

### PR DESCRIPTION
**Pull request by matharden**
## User story

[As a monies clerk I need to be introduced to the Tool Tour so that I can understand how to use the digital cashbook](https://www.pivotaltracker.com/story/show/97105820) - 97105820

## Adds the following dependancies

- uses [Hopscotch](https://github.com/linkedin/hopscotch) feature tour by LinkedIn which supports IE8
- augmented by [Hopscotch-Highlight](https://github.com/ministryofjustice/hopscotch-highlight) which masks each feature
- uses [js-cookie](https://github.com/js-cookie/js-cookie) for handling state

**Note:** this merges into internal-template (#33).

### CODE REVIEW

File: mtp_cashbook/assets-src/javascripts/app.js [hunk 1]
```diff
@@ -17,5 +17,72 @@
   checks.on('change', function() {
     var check = $(this);
     check.closest('tr')[check.is(':checked') ? 'addClass' : 'removeClass']('is-selected');
-  })
+  });
+
+  var tourBase = {
```
Comment (teneightfive) [inline mtp_cashbook/assets-src/javascripts/app.js]: Can this be extracted out into a separate module using heisenbergJS style?

File: bower.json [hunk 2]
```diff
@@ -11,6 +11,10 @@
   ],
   "dependencies": {
     "lodash": "~2.4.1",
-    "mojular": "matharden/mojular#internal"
+    "mojular": "matharden/mojular#internal2",
```
Comment (SteveMarshall) [inline bower.json]: Is this branch going to be moved somewhere less @matharden-centric? @teneightfive?

File: mtp_cashbook/assets-src/javascripts/app.js [hunk 3]
```diff
@@ -17,5 +17,72 @@
...
+      var stepNo = hopscotch.getCurrStepNum();
+
+      hopscotch.highlight.remove();
+
+      Cookies.set('hopscotch.state.' + tour.id, 'dismissed');
+
+      if (tour.steps[stepNo].dismissTo) {
+        window.setTimeout(function(){
+          hopscotch.startTour(tour, tour.steps[stepNo].dismissTo);
+        }, 300);
```
Comment (SteveMarshall) [inline mtp_cashbook/assets-src/javascripts/app.js]: Should this be a parameter or something?

File: mtp_cashbook/assets-src/javascripts/app.js [hunk 4]
```diff
@@ -17,5 +17,72 @@
...
+          $('.hopscotch-bubble').fadeOut(300, function(){
+            hopscotch.endTour(false);
+          });
+        }, 5000);
+      }
+
+      if (tour.steps[stepNo].target === 'tour-intro') {
+        $('.hopscotch-bubble-arrow-container').remove();
+        $('.hopscotch-nav-button').text('Yes please');
+        $('.hopscotch-bubble-close').text('Not now');
```
Comment (SteveMarshall) [inline mtp_cashbook/assets-src/javascripts/app.js]: Can we pull these blobs of text from somewhere Django-translatable?

**Comment by SteveMarshall** [comment]
Can you rebase this against `master` (or re-open with a new PR) so we can see what's already been done by #33?

File: mtp_cashbook/assets-src/javascripts/app.js (hunk 4 above)
Comment (matharden) [inline mtp_cashbook/assets-src/javascripts/app.js]: That would be good. Although I can't think how to do that right now. Perhaps one for the new dev?

File: mtp_cashbook/assets-src/javascripts/app.js (hunk 1 above)
Comment (matharden) [inline mtp_cashbook/assets-src/javascripts/app.js]: I'm not sure that really fits into the module pattern of Heisenburg. It's simply settings in an object literal for an external component.

**Comment by matharden** [comment]
Replaced by #36

**Pull request closed without merging**